File: models/category.py
class Category:
    def __init__(self, category_id: str, name: str, icon: str = "", description: str = ""):
        self.category_id = category_id
        self.name = name
        self.icon = icon
        self.description = description

    def to_dict(self) -> dict:
        return {
            "id": self.category_id,
            "name": self.name,
            "icon": self.icon,
            "description": self.description,
        }

    @classmethod
    def from_dict(cls, data: dict) -> "Category":
        return cls(
            category_id=data.get("id", ""),
            name=data.get("name", ""),
            icon=data.get("icon", ""),
            description=data.get("description", ""),
        )

    def __repr__(self):
        return f"<Category: {self.name}>"

File: flutter_app/lib/models/category.dart
class Category {
  final String id;
  final String name;
  final String icon;
  final String description;

  const Category({
    required this.id,
    required this.name,
    this.icon = '',
    this.description = '',
  });

  Map<String, dynamic> toMap() => {
        'id': id,
        'name': name,
        'icon': icon,
        'description': description,
      };

  factory Category.fromMap(Map<String, dynamic> data) {
    return Category(
      id: (data['id'] ?? '').toString(),
      name: (data['name'] ?? '').toString(),
      icon: (data['icon'] ?? '').toString(),
      description: (data['description'] ?? '').toString(),
    );
  }

  @override
  String toString() => '<Category: $name>';
}